src/backend/aspen/workflows/nextclade/save.py

```diff
@@ -92,44 +94,43 @@ def cli(
+                    SampleMutation.mutations_caller == MutationsCaller.NEXTCLADE,
                 )
-                mutation = session.execute(existing_mutation_q).scalars().one_or_none()
+            )
+            mutation = session.execute(existing_mutation_q).scalars().one_or_none()
 
-                if mutation is None:
-                    mutation = SampleMutation(
-                        sample=sample,
-                        mutations_caller=MutationsCaller.NEXTCLADE,
-                        substitutions=row["substitutions"],
-                        insertions=row["insertions"],
-                        deletions=row["deletions"],
-                        aa_substitutions=row["aaSubstitutions"],
-                        aa_insertions=row["aaInsertions"],
-                        aa_deletions=row["aaDeletions"],
-                        reference_sequence_accession=dataset_info["accession"],
-                    )
-                else:
-                    mutation.substitutions = row["substitutions"]
-                    mutation.insertions = row["insertions"]
-                    mutation.deletions = row["deletions"]
-                    mutation.aa_substitutions = row["aaSubstitutions"]
-                    mutation.aa_insertions = row["aaInsertions"]
-                    mutation.aa_deletions = row["aaDeletions"]
-                    mutation.reference_sequence_accession = dataset_info["accession"]
-                session.add(mutation)
-
-            # If run was invalid, we do not set any info about the lineage.
-            # Additionally, if SC2 (covid) we use Pangolin, not Nextclade.
-            if is_result_valid and pathogen_slug != "SC2":
-                lineage = get_lineage_from_row(row)
+            if mutation is None:
+                mutation = SampleMutation(
+                    sample=sample,
+                    mutations_caller=MutationsCaller.NEXTCLADE,
+                    substitutions=row["substitutions"],
+                    insertions=row["insertions"],
+                    deletions=row["deletions"],
+                    aa_substitutions=row["aaSubstitutions"],
+                    aa_insertions=row["aaInsertions"],
+                    aa_deletions=row["aaDeletions"],
+                    reference_sequence_accession=dataset_info["accession"],
+                )
+            else:
+                mutation.substitutions = row["substitutions"]
+                mutation.insertions = row["insertions"]
+                mutation.deletions = row["deletions"]
+                mutation.aa_substitutions = row["aaSubstitutions"]
+                mutation.aa_insertions = row["aaInsertions"]
+                mutation.aa_deletions = row["aaDeletions"]
+                mutation.reference_sequence_accession = dataset_info["accession"]
+            session.add(mutation)
+
+            # If SC2 (covid) we use Pangolin, not Nextclade.
+            if pathogen_slug != "SC2":
```

Comment:
- This logic could be inverted to reduce nesting, something more like `if pathogen_slug == sc2: return`
- This could also use a feature flag instead of a hardcoded pathogen